### PR TITLE
win: fix some MinGW warnings

### PR DESCRIPTION
**Pull request by escherstair**
fix some warnings from MinGW compiler while compiling on win
See #2790

### CODE REVIEW

File: src/win/util.c [hunk 1]
```diff
@@ -1804,7 +1804,9 @@ int uv_os_uname(uv_utsname_t* buffer) {
     pRtlGetVersion(&os_info);
   } else {
     /* Silence GetVersionEx() deprecation warning. */
+    #if _MSC_VER
```
Comment (addaleax) [inline src/win/util.c:1807]: Elsewhere we use `#ifdef`, I think:

```suggestion
    #ifdef _MSC_VER
```

**Comment by escherstair** [comment]
Is the failure related to my fix?

**Comment by bnoordhuis** [comment]
@libuv/windows This needs a review.

**Comment by bzoz** [comment]
LGTM as those are mostly style changes. CI: https://ci.nodejs.org/view/libuv/job/libuv-test-commit/1928/

File: src/win/util.c [hunk 2]
```diff
@@ -30,12 +30,14 @@
 #include "uv.h"
 #include "internal.h"
 
+// clang-format off
```
Comment (Turbinya) [inline src/win/util.c:33]:
```suggestion
/* clang-format off */
```

File: src/win/util.c [hunk 3]
```diff
@@ -30,12 +30,14 @@
...
 #include <winsock2.h>
 #include <winperf.h>
 #include <iphlpapi.h>
 #include <psapi.h>
 #include <tlhelp32.h>
 #include <windows.h>
+// clang-format on
```
Comment (Turbinya) [inline src/win/util.c:40]:
```suggestion
/* clang-format on */
```

**Comment by vtjnash** [comment]
Thanks! Can you rebase and squash these (as applicable), and update their commit messages, per the contributing file guidelines, to have the following format / content:
```
win,nfc: <rest of title>

PR-URL: https://github.com/libuv/libuv/pull/2854
Reviewed-By: Anna Henningsen <anna@addaleax.net>
Reviewed-By: Bartosz Sosnowski <bartosz@janeasystems.com>
Reviewed-By: Jameson Nash <vtjnash@gmail.com>
```

**Comment by escherstair** [comment]
I rebased but I have problems in squashing and changing commit labels

**Comment by escherstair** [comment]
I rebased over v1.x branch, squashed the commits and updated the messages

**Pull request merged**